Upstream-Status: Pending

Don't always define LONG64 for AMD64
    
X32 defines __amd64__/amd64 with 32bit long.  We should simply check
__LP64__ before defining LONG64 without checking __amd64__/amd64.

This fixes compilation with x32 toolchain.

Received this patch from H.J. Lu <hjl.tools@gmail.com>
Signed-off-by: Nitin A Kamble <nitin.a.kamble@intel.com> 2011/12/1

Index: xproto-7.0.31/Xmd.h
===================================================================
--- xproto-7.0.31.orig/Xmd.h
+++ xproto-7.0.31/Xmd.h
@@ -62,7 +62,6 @@ SOFTWARE.
      defined(__ia64__) || defined(ia64) || \
      defined(__sparc64__) || \
      defined(__s390x__) || \
-     defined(__amd64__) || defined(amd64) || \
      defined(__powerpc64__)
 #  if !defined(__ILP32__) /* amd64-x32 is 32bit */
 #   define LONG64				/* 32/64-bit architecture */
